icechunk-python/tests/test_can_read_old.py

```diff
@@ -195,6 +197,9 @@ async def test_icechunk_can_read_old_repo() -> None:
         1:
     ]
 
+    with pytest.raises(ValueError, match="ref not found"):
```

Comment:
Sounds we make a new subclass of IcechunkError for this while working on this?

Reply:
I think our whole error hierarchy needs a revamp. Let me delay this until we do that rfactoring.